### PR TITLE
Print error message if GPU is not present.

### PR DESCRIPTION
**Pull request by indhub**

### CODE REVIEW

**Comment by piiswrong** [comment]
explain this in text instead

**Comment by indhub** [comment]
@bhavinthaker wanted an error message instead of stacktrace when GPU is not present. 

Bhavin - see comment from @piiswrong. Let me know if a comment saying GPU is required is sufficient.

**Comment by bhavinthaker** [comment]
@piiswrong Why do we need a stack trace for an error condition which is as simple as GPU not present when specified? A simple error message and no stack trace improves product usability.

**Comment by piiswrong** [comment]
Code in tutorial is supposed to be easy to read. We shouldn't make user ready through irrelevant boiler-plate code.

A note in the beginning saying if you don't have gpu an error will be raised is better. Instead of running and seeing error message users would simply stop reading.

Use a `.. note:` note section to make it more visible

**Comment by bhavinthaker** [comment]
@piiswrong I see your point. I agree -- The tutorial needs to be easy to read. I am fine with your suggestion.

**Comment by madjam** [comment]
+1 to what @piiswrong said. Adding this code to the tutorial interferes with the flow.
However, from the error message one sees, it is not immediately obvious that there is no GPU.
A suggestion would be to (a) add text here that says this will fail if no GPU exists (b) simplify the error message in mxnet core. Perhaps, add a assert that checks GPU count > 0?

File: docs/tutorials/basic/ndarray.md [hunk 1]
```diff
@@ -279,6 +279,8 @@ can cause all computations to run on GPU 0 by using context `mx.gpu(0)`, or
 simply `mx.gpu()`. When we have access to two or more GPUs, the 2nd GPU is
 represented by `mx.gpu(1)`, etc.
 
+*Note: To use NDArray's GPU features, a GPU must be present and MXNet must be compiled with USE_CUDA=1*
```
Comment (mli) [inline docs/tutorials/basic/ndarray.md]: why we need it? there is already 

> When MXNet is compiled with flag USE_CUDA=1 and the machine has at least one NVIDIA GPU, 

in the paragraph

File: docs/tutorials/basic/symbol.md [hunk 2]
```diff
@@ -320,11 +320,13 @@ print 'number of outputs = %d\nthe first output = \n%s' % (
...
 We can evaluate the same symbol on GPU with different data.
 
+*Note: To use the GPU functionalities, a GPU must be present and MXNet must be compiled with USE_CUDA=1*
+
 ```python
-ex_gpu = c.bind(ctx=mx.gpu(), args={'a' : mx.nd.ones([3,4], mx.gpu())*2,
-                                    'b' : mx.nd.ones([3,4], mx.gpu())*3})
-ex_gpu.forward()
-ex_gpu.outputs[0].asnumpy()
+    ex_gpu = c.bind(ctx=mx.gpu(), args={'a' : mx.nd.ones([3,4], mx.gpu())*2,
```
Comment (mli) [inline docs/tutorials/basic/symbol.md]: don't add empty spaces at the beginning. indent matters for python

**Comment by mli** [comment]
@bhavinthaker the tutorial already explains you should have GPU installed and be compiled with USE_CUDA=1 to run on GPU. I don't think we need to double emphasize it. I'm going to close the PR, but please feel free to reopen it.

**Pull request closed without merging**